.release-notes/4299.md

```diff
@@ -0,0 +1,3 @@
+## Fix vague error message when it is not safe to write
+
+Added left side type to a "not safe to write" error message.
```

Comment:
This is likely to be lacking in meaning for a reader of the release notes who doesn't have your context.

Please include what the error message used to be, why that was vague, and how the new version of the message fixes the vagueness.